File: src/test/java/org/dicio/skill/standard/word/DiacriticsInsensitiveWordTest.java
package org.dicio.skill.standard.word;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import org.dicio.skill.util.WordExtractor;
import org.junit.Test;

public class DiacriticsInsensitiveWordTest {


    private DiacriticsInsensitiveWord diw(final String value) {
        return new DiacriticsInsensitiveWord(WordExtractor.nfkdNormalizeWord(value), 0);
    }

    private void assertMatches(final String value, final String... inputWords) {
        final DiacriticsInsensitiveWord diacriticsInsensitiveWord = diw(value);
        for (final String word : inputWords) {
            assertTrue(value + " should match " + word
                            + " (normalized: " + WordExtractor.nfkdNormalizeWord(word) + ")",
                    diacriticsInsensitiveWord.matches(word, WordExtractor.nfkdNormalizeWord(word)));
        }
    }

    private void assertNotMatches(final String value, final String... inputWords) {
        final DiacriticsInsensitiveWord diacriticsInsensitiveWord = diw(value);
        for (final String word : inputWords) {
            assertFalse(value + " should not match " + word
                            + " (normalized: " + WordExtractor.nfkdNormalizeWord(word) + ")",
                    diacriticsInsensitiveWord.matches(word, WordExtractor.nfkdNormalizeWord(word)));
        }
    }


    @Test
    public void testMatches() {
        assertMatches("hello", "hèllo", "hellò", "héllò");
        assertMatches("ùlìc", "ùliç", "ulic", "ulìc");
    }

    @Test
    public void testNotMatches() {
        assertNotMatches("hello", "ħello", "hèłlo", "ħéłłò");
        assertNotMatches("ciao", "ciau", "chiao", "hiao", "chao");
    }
}

File: src/test/java/org/dicio/skill/standard/word/DiacriticsInsensitiveWordTest.kt
package org.dicio.skill.standard.word

import org.dicio.skill.util.WordExtractor.nfkdNormalizeWord
import org.junit.Assert
import org.junit.Test

class DiacriticsInsensitiveWordTest {
    private fun diw(value: String): DiacriticsInsensitiveWord {
        return DiacriticsInsensitiveWord(nfkdNormalizeWord(value), 0)
    }

    private fun assertMatches(value: String, vararg inputWords: String) {
        val diacriticsInsensitiveWord = diw(value)
        for (word in inputWords) {
            Assert.assertTrue(
                value + " should match " + word
                        + " (normalized: " + nfkdNormalizeWord(word) + ")",
                diacriticsInsensitiveWord.matches(word, nfkdNormalizeWord(word))
            )
        }
    }

    private fun assertNotMatches(value: String, vararg inputWords: String) {
        val diacriticsInsensitiveWord = diw(value)
        for (word in inputWords) {
            Assert.assertFalse(
                value + " should not match " + word
                        + " (normalized: " + nfkdNormalizeWord(word) + ")",
                diacriticsInsensitiveWord.matches(word, nfkdNormalizeWord(word))
            )
        }
    }


    @Test
    fun testMatches() {
        assertMatches("hello", "hèllo", "hellò", "héllò")
        assertMatches("ùlìc", "ùliç", "ulic", "ulìc")
    }

    @Test
    fun testNotMatches() {
        assertNotMatches("hello", "ħello", "hèłlo", "ħéłłò")
        assertNotMatches("ciao", "ciau", "chiao", "hiao", "chao")
    }
}
